### PR TITLE
Sets the curl proxy ssl verify options to the values of the host configuration options

### LINKED ISSUE #1616: Cannot skip CA verification when using CURL input plugin
### Describe the bug
When using curl as an input plugin, block variables "verify_peer", "verify_host", and "cacert" don't seem to have any affect.

## Expected Behavior
I have the following setting:
```
input {
        plugin "curl"
        proxy "https://my-host"
        proxy_user "me"
        proxy_password "mypass"
        verify_peer "no"
        verify_host "no"
}
```
I expect certificate verification to be skipped.

## Actual Behavior
However, the following error always occurs:
```
CURL failed: server certificate verification failed. CAfile: none CRLfile: none
```

## Version
Music Player Daemon 0.23.9 (0.23.9)
Copyright 2003-2007 Warren Dukes <warren.dukes@gmail.com>
Copyright 2008-2021 Max Kellermann <max.kellermann@gmail.com>
This is free software; see the source for copying conditions.  There is NO
warranty; not even MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Database plugins:
 simple proxy upnp

Storage plugins:
 local smbclient udisks nfs curl

Neighbor plugins:
 smbclient upnp udisks

Decoders plugins:
 [mad] mp3 mp2
 [mpg123] mp3
 [vorbis] ogg oga
 [oggflac] ogg oga
 [flac] flac
 [opus] opus ogg oga
 [sndfile] wav aiff aif au snd paf iff svx sf voc w64 pvf xi htk caf sd2
 [dsdiff] dff
 [dsf] dsf
 [hybrid_dsd] m4a
 [faad] aac
 [mpcdec] mpc
 [wavpack] wv
 [openmpt] mptm mod s3m xm it 669 amf ams c67 dbm digi dmf dsm dtm far imf ice j2b m15 mdl med mms mt2 mtm nst okt plm psm pt36 ptm sfx sfx2 st26 stk stm stp ult wow gdm mo3 oxm umx xpk ppm mmcmp
 [modplug] 669 amf ams dbm dfm dsm far it med mdl mod mtm mt2 okt s3m stm ult umx xm
 [mikmod] amf dsm far gdm imf it med mod mtm s3m stm stx ult uni xm
 [sidplay] sid mus str prg P00
 [wildmidi] mid
 [fluidsynth] mid
 [adplug] amd d00 hsc laa rad raw sa2
 [ffmpeg] 16sv 3g2 3gp 4xm 8svx aa3 aac ac3 adx afc aif aifc aiff al alaw amr anim apc ape asf atrac au aud avi avm2 avs bap bfi c93 cak cin cmv cpk daud dct divx dts dv dvd dxa eac3 film flac flc fli fll flx flv g726 gsm gxf iss m1v m2v m2t m2ts m4a m4b m4v mad mj2 mjpeg mjpg mka mkv mlp mm mmf mov mp+ mp1 mp2 mp3 mp4 mpc mpeg mpg mpga mpp mpu mve mvi mxf nc nsv nut nuv oga ogm ogv ogx oma ogg omg opus psp pva qcp qt r3d ra ram rl2 rm rmvb roq rpl rvc shn smk snd sol son spx str swf tak tgi tgq tgv thp ts tsp tta xa xvid uv uv2 vb vid vob voc vp6 vmd wav webm wma wmv wsaud wsvga wv wve rtp:// rtsp:// rtsps://
 [gme] ay gbs gym hes kss nsf nsfe rsn sap spc vgm vgz
 [pcm]

Filters:
 libsamplerate soxr

Tag plugins:
 id3tag

Output plugins:
 shout null fifo sndio pipe alsa ao oss openal pipewire pulse jack httpd snapcast recorder

Encoder plugins:
 null vorbis opus lame wave flac

Archive plugins:
 [bz2] bz2
 [zzip] zip
 [iso] iso

Input plugins:
 file io_uring archive alsa curl ffmpeg smbclient nfs mms cdio_paranoia

Playlist plugins:
 extm3u m3u pls xspf asx rss soundcloud flac cue embcue

Protocols:
 file:// alsa:// cdda:// ftp:// ftps:// gopher:// hls+http:// hls+https:// http:// https:// mms:// mmsh:// mmst:// mmsu:// nfs:// rtmp:// rtmps:// rtmpt:// rtmpts:// rtp:// rtsp:// rtsps:// scp:// sftp:// smb:// srtp://

Other features:
 avahi dbus udisks epoll icu inotify ipv6 systemd tcp un

## Configuration
```
music_directory	 "https://my-host/music"

input {
        plugin "curl"
        proxy "https://my-host"
        proxy_user "me"
        proxy_password "mypass"
        verify_peer "no"
        verify_host "no"
}

....other settings....
```

## Log
```
CURL failed: server certificate verification failed. CAfile: none CRLfile: none
```

### PR DESCRIPTION
**Pull request by jcorporation**
This fixes #1616

Or do you want two new configuration options for the curl plugin?

### CODE REVIEW

File: src/lib/curl/Request.hxx [hunk 1]
```diff
@@ -122,6 +122,14 @@ public:
 		easy.SetVerifyPeer(value);
 	}
 
+	void SetProxyVerifyHost(bool value) {
+		easy.SetOption(CURLOPT_PROXY_SSL_VERIFYHOST, value == 1 ? 2L : 0L);
```
Comment (MaxKellermann) [inline src/lib/curl/Request.hxx:126]: Don't compare a bool with an int!

**Comment by MaxKellermann** [comment]
I'm not sure if that should be a separate option - I guess if you disable verify, you're completely lost already anyway? Is it helpful to verify the proxy but not the final server?

File: src/lib/curl/Request.hxx (hunk 1 above)
Comment (jcorporation) [reply src/lib/curl/Request.hxx]: Fixed

**Comment by MaxKellermann** [comment]
`value == true` is a complicated way of saying `value` ;-)

**Comment by jcorporation** [comment]
Reading https://curl.se/libcurl/c/CURLOPT_PROXY_SSL_VERIFYHOST.html
This option must be set to 2L not 1 (true), if I do not missread the documentation.

**Comment by MaxKellermann** [comment]
Cherry-picked to v0.23.x 7ab0dfc8ce56dacee9206b4c429a6c4bd94394b4

**Comment by MaxKellermann** [comment]
> This option must be set to 2L not 1 (true), if I do not missread the documentation.

True, and that's what your code does. The `bool` check is on the other side of the expression and is unrelated to that.

**Pull request closed without merging**